cmd/server/pactasrv/user.go

```diff
@@ -4,72 +4,83 @@ import (
+	user, err := s.DB.User(s.DB.NoTxn(ctx), pacta.UserID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to retrieve user", zap.Error(err))
+	}
+	result, err := conv.UserToOAPI(user)
+	if err != nil {
+		return nil, err
+	}
+	return api.FindUserById200JSONResponse(*result), nil
 }
 
 // Updates user properties
 // (PATCH /user/{id})
 func (s *Server) UpdateUser(ctx context.Context, request api.UpdateUserRequestObject) (api.UpdateUserResponseObject, error) {
 	// TODO(#12) Implement Authorization
-	return nil, oapierr.NotImplemented("updateUser")
+	mutations := []db.UpdateUserFn{}
+	if request.Body.Admin != nil {
+		mutations = append(mutations, db.SetUserAdmin(*request.Body.Admin))
+	}
+	if request.Body.SuperAdmin != nil {
+		mutations = append(mutations, db.SetUserSuperAdmin(*request.Body.SuperAdmin))
+	}
+	if request.Body.Name != nil {
+		mutations = append(mutations, db.SetUserName(*request.Body.Name))
+	}
+	if request.Body.PreferredLanguage != nil {
+		lang, err := pacta.ParseLanguage(string(*request.Body.PreferredLanguage))
+		if err != nil {
+			return nil, oapierr.BadRequest("invalid language", zap.Error(err))
+		}
+		mutations = append(mutations, db.SetUserPreferredLanguage(lang))
+	}
+	err := s.DB.UpdateUser(s.DB.NoTxn(ctx), pacta.UserID(request.Id), mutations...)
+	if err != nil {
+		return nil, oapierr.Internal("failed to update user", zap.Error(err))
+	}
+	return api.UpdateUser204Response{}, nil
 }
 
 // Deletes a user by ID
 // (DELETE /user/{id})
 func (s *Server) DeleteUser(ctx context.Context, request api.DeleteUserRequestObject) (api.DeleteUserResponseObject, error) {
 	// TODO(#12) Implement Authorization
-	return nil, oapierr.NotImplemented("deleteUser")
+	err := s.DB.DeleteUser(s.DB.NoTxn(ctx), pacta.UserID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to delete user", zap.Error(err))
+	}
+	return api.DeleteUser204Response{}, nil
 }
 
 // Returns the logged in user
 // (GET /user/me)
 func (s *Server) FindUserByMe(ctx context.Context, request api.FindUserByMeRequestObject) (api.FindUserByMeResponseObject, error) {
-	token, _, err := jwtauth.FromContext(ctx)
-	if err != nil {
-		return nil, oapierr.Unauthorized("without authorization token")
-	}
-	if token == nil {
-		return nil, oapierr.Unauthorized("without authorization token")
-	}
-	emailsClaim, ok := token.PrivateClaims()["emails"]
-	if !ok {
-		return nil, oapierr.Unauthorized("without email claim in token")
-	}
-	emails, ok := emailsClaim.([]string)
-	if !ok || len(emails) == 0 {
-		return nil, oapierr.Internal("couldn't parse email claim in token")
-	}
-	// TODO(#18) Handle Multiple Emails in the Token Claims gracefully
-	if len(emails) > 1 {
-		return nil, oapierr.BadRequest("multiple emails in token")
-	}
-	email := emails[0]
-	canonical, err := pacta.CanonicalizeEmail(email)
+	// Looking for how users get created or populated into the context?
+	// It's in the HTTP handler for adding a user to the context in main.go
```

Comment:
After you move everything into a `session` package, it'll be more obvious and this comment can be removed.

Reply:
Will Remove.